src/plopp/core/limits.py

```diff
@@ -1,21 +1,60 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
 
-from typing import Literal, Tuple
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Literal, Optional, Tuple
 
 import numpy as np
-from scipp import Variable, scalar
+import scipp as sc
+
+from .utils import merge_masks
+
+
+@dataclass
+class BoundingBox:
+    """
+    A bounding box in 2D space.
+    """
+
+    xmin: Optional[float] = None
+    xmax: Optional[float] = None
+    ymin: Optional[float] = None
+    ymax: Optional[float] = None
+
+    def union(self, other: BoundingBox) -> BoundingBox:
+        """
+        Return the union of this bounding box with another one.
+        """
+        return BoundingBox(
+            xmin=min(self.xmin or np.inf, other.xmin or np.inf),
```

Comment:
Wouldn't this break if `self.xmin == 0`? I think we need to explicitly check for `None` here even if it is more verbose. Or perhaps we can set the defaults already in the constructor?